gabbi/case.py

```diff
@@ -190,9 +190,12 @@ def load_data_file(self, filename):
 
     def _assert_response(self):
         """Compare the response with expected data."""
+        handlers = self.response_handlers
+        content_handler = self.get_content_handler(self.content_type)
+        if content_handler:
+            handlers.append(content_handler)
```

Comment:
This isn't the right fix for this. Unfortunately, we're going to need to dig a bit to figure out what the right fix is. The problem will be happening somewhere in the `_register` handling in the base `ResponseHandler`.

Part of the reason this is wrong is that this method is supposed to be using response handlers only (of which content handlers are a subclass, but not all responsehandlers are content handlers) so if you have to go looking in get_content_handler than a handler has not _registered properly, or has been clobbered somehow.

I haven't got the cycles to dig into it today.

Reply:
I added a class attribute to help with this.